### DISCUSSION TITLE
Rust file watcher apparently leaks

### DISCUSSION THREAD

**Discussion by hubertp** [enso-org/enso#7654] [Work In Progress - GUI]
My typical setup is that in one console I start GUI and in the other I typically iterate over backend.
Since I'm not really doing much on the IDE side I wouldn't expect it to affect my setup in any way.
Yet, after a long session I started getting kernel inotify watch limit failures all over the place and things became really slow over time.

That's not surprising given how much of files were being tracked by cargo watcher:
```
22921     1803418 2182600  1000 /home/hubert/opt/idea-IC-223.8617.56/bin/fsnotifier
32194     2029496 2029599  1000 /home/hubert/.cargo/bin/cargo-watch watch --ignore README.md -- /home/hubert/work/repos/enso/enso/target/enso-build/buildscript/enso-build-cli --skip-version-check --cache-path /home/hubert/.local/share/.enso-ci/cache --up
```
(the first number is the number of open files).

I can understand that my editor is somehow opening a lot of files. But file watcher in enso is a bit too much. Any ideas?

**Comment by vitvakatu** [comment]
That's interesting, given that we have only `5034` files in the repository, including Rust, Scala and Java sources, various configs and markdown documentation files.

EDIT: the amount of files in general is much, much larger. I have around 200'000 at the moment.

**Comment by kazcw** [comment]
We have an [alternate watcher](https://github.com/enso-org/enso/blob/c80b8624a84faaa977776e73257f3aec369f4a99/docs/CONTRIBUTING.md?plain=1#L253). AFAIK we have fewer problems with it than `cargo-watch`, though I haven't tested it myself (I usually build manually).
